django/db/models/base.py

```diff
@@ -75,6 +75,18 @@ def __new__(cls, name, bases, attrs, **kwargs):
         classcell = attrs.pop('__classcell__', None)
         if classcell is not None:
             new_attrs['__classcell__'] = classcell
+        # Fix related to bug #28198.
+        # Remove duplicate public attributes from parent classes
+        attr_per_class = {}
+        for parent in parents:
+            attr_per_parent = {}
+            for key in attrs.keys():
+                if hasattr(parent, key) and not key.startswith('__') and not hasattr(getattr(parent, key),
+                                                                                     '__dict__'):
+                    attr_per_parent[key] = getattr(parent, key)
+                    delattr(parent, key)
+            if len(attr_per_parent.keys()) > 0:
```

Comment:
`if attr_per_parent:` is simpler -- does the current implementation have an advantage over that? (It looks like it requires more computation.) Same comment below.